app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @total_cost = @recipe.total_cost
```

Comment:
It's totally okay to have an additional @total_cost instance variable. But for the future, it's also okay to call `@recipe.total_cost` from the view as well. Calling a model method from the view is not an MVC infraction, but including the actual AR query in the view (like `recipe.ingredients.sum(:cost)`) is what we want to avoid.

Reply:
Just to be clear, what you did here is not wrong at all. It's just another way to do it! I think students, sometimes, are afraid of calling model methods from the view but that's completely okay. You're encapsulating the AR query at the model level so that you can call the model method in controllers or views.